### DISCUSSION TITLE
ctx is empty, ctx.session is undefined

### DISCUSSION THREAD

**Discussion by jrydberg** [blitz-js/blitz#2321] [Q&A]
First time trying out blitz.js - looks awesome.  But ran into a little snag when trying to implement login using firebase. It looks like `Ctx` is `{}` in my mutation. Which means that I cannot create a new session since `ctx.session` is undefined.  Anyone have any clue to why this could happen?

Could be related to that I'm playing around with a `blitz.config.ts`

**Comment by flybayer** [comment]
So you removed `sessionMiddleware` and are trying to use Firebase instead? `ctx.session` is added by `sessionMiddleware`. So if you don't use that, you will need to implement your own `ctx.session`. Probably you are better off to use Blitz sessions instead of firebase :)

Is the Blitz built in sessions not working for something that makes you want to use Firebase?

**Reply by jrydberg** [reply]
I was using the wrong version of Blitz :).  Now with 0.35 that properly supports `blitz.config.ts` things work a lot better.

I'm using Firestore for my application already, and I would like to stick to using one database.  The builtin session mgmt seems sweet, but I don't wanna have another database just for that.

**Reply by flybayer** [reply]
Ok, then another option is customizing sessionMiddleware to use the firestore database for sessions: https://blitzjs.com/docs/session-management#customize-session-persistence-and-database-access
